### PR TITLE
[enhancement] conform to sklearn1.6 ```validate_data``` API change

### PR DESCRIPTION
**Pull request by icfaust**
### Description

There was an API change in sklearn which moved ```_validate_data``` from ```BaseEstimator``` to ```sklearn.utils.validation``` meaning that sklearnex must do some API call changes scikit-learn/scikit-learn#29696.

This also adds testing to ```test_common.py``` which regulates the use of ```validate_data``` and ```check_feature_names``` which now enforces a rule that the check for feature names will occur via the ```validate_data``` call.

_Add a comprehensive description of proposed changes_

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by icfaust** [comment]
Not sure why SVC/NuSVC requires ensure_2d=False on predict, but when True is causes sklearn conformance failures....

**Review by Alexsandruss — APPROVED** [review]
